pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -300,29 +300,17 @@ func (m *Matrix) validateCombinationsCount(ctx context.Context) (errs *apis.Fiel
 	return errs
 }
 
-// validateParams validates the type of Parameter for Matrix.Params and Matrix.Include.Params
-// Matrix.Params must be of type array. Matrix.Include.Params must be of type string.
-// validateParams also validates Matrix.Params for a unique list of params
-// and a unique list of params in each Matrix.Include.Params specification
-func (m *Matrix) validateParams() (errs *apis.FieldError) {
+// validateUniqueParams validates there are no duplicate parameters
+// within Matrix.Params or Matrix.Include Params
+func (m *Matrix) validateUniqueParams() (errs *apis.FieldError) {
 	if m != nil {
 		if m.HasInclude() {
 			for i, include := range m.Include {
 				errs = errs.Also(include.Params.validateDuplicateParameters().ViaField(fmt.Sprintf("matrix.include[%d].params", i)))
-				for _, param := range include.Params {
```

Comment:
I don't really understand why this validation is being moved from apis to the reconciler

Reply:
@lbernick The validation in the reconciler happens after parameter substitution when the parameter type will be array vs here it is hard to validate the underlying type it is referencing since the type is a string.

Reply:
ah thanks for the explanation! Is it possible that there could be duplicate parameters after parameter substitution happens, and that unique param validation should also be moved to the reconciler?

Reply:
I think that's a valid point  I moved the uniqueParam validation to the reconciler after validating types.